File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/TableRequestDetails.java
package kz.greetgo.sandbox.controller.model;

public class TableRequestDetails {
  public int pageIndex =  1;
  public int pageSize =  3;
  public SortColumn sortBy =  SortColumn.NONE;
  public SortDirection sortDirection =  SortDirection.ASC;
  public String filter =  "";
}

File: sandbox.client/front/ts/model/TableRequestDetails.ts
import {SortDirection} from "./SortDirection";

export class TableRequestDetails {
  public pageIndex: number = 1;
  public pageSize: number = 3;
  public sortBy: string = '';
  public sortDirection: SortDirection = SortDirection.ASC;
  public filter: string = '';


  constructor(pageIndex: number, pageSize: number, sortBy: string, sortDirection: SortDirection, filter: string) {
    this.pageIndex = pageIndex;
    this.pageSize = pageSize;
    this.sortBy = sortBy;
    this.sortDirection = sortDirection;
    this.filter = filter;
  }

  toString() {
    return "PageIndex:" + this.pageIndex + ", PageSize:" + this.pageSize
      + ", SortBy:" + this.sortBy + ", Sort:" + this.sortDirection + ", Filter:" + this.filter;
  }
}